docs/docs/api/resource-management.md

```diff
@@ -35,14 +35,18 @@ this would lose a lot of the benefits of using asynchronity and we might almost
 We can do better, retaining concurrency and not leaking resources, by using `.using`:
 
 ```js
-var using = Promise.using;
-
-using(getConnection(),
-      fs.readFileAsync("file.sql", "utf8"), function(connection, fileContents) {
-    return connection.query(fileContents);
-}).then(function() {
-    console.log("query successful and connection closed");
-});
+function doStuff() {
+    return Promise.using([
+        connectionPool.getConnectionAsync(),
+        fs.readFileAsync("file.sql", "utf8")
+    ]), function(connection, fileContents) {
+        return connection.query(fileContents);
+    }).then(function() {
+        console.log("query successful and connection closed");
+    }).finally(function() {
+        connection.close();
```

Comment:
Its not necessary to dispose of resources with `Promise.using`. As soon as the promise returned within the closure passed to `using` resolves, the disposer for the connection will be immediately invoked. `Using` will then wait for it to call the `close()` method (waiting for any promises it might return), then return the original result.

Basically `Promise.using` is a safe, automated way to ensure that finalizers are called. 

The docs may not be quite clear in stating this - hopefully we can improve them together.